matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/store/db/RealmCryptoStore.kt

```diff
@@ -1815,4 +1822,44 @@ internal class RealmCryptoStore @Inject constructor(
             // Can we do something for WithHeldSessionEntity?
         }
     }
+
+    private var cryptoStoreAggregator: CryptoStoreAggregator? = null
+    override fun onSyncWillProcess() {
+        if (cryptoStoreAggregator != null) {
+            Timber.e("cryptoStoreAggregator is not null...")
+        }
+        cryptoStoreAggregator = CryptoStoreAggregator()
```

Comment:
I usually don't like when we add cache/buffer directly in the RealmStore.
We used to do it for inbound session and it ended up a bit hard to manage.
I'd prefer to an abstraction on top of the store (BatchedStoreHelper or something); that calls the store that is only doing storing (a bit like the session cache was moved to InboundGroupSessionStore).
But maybe too much?

Reply:
OK, reworked.